manifests/master.pp

```diff
@@ -171,69 +168,144 @@
-  Boolean                         $environment_class_cache_enabled = true,
-  Optional[Pattern['^\d+\.\d+$']] $compat_version                  = undef,
-  Enum['off', 'jit', 'force']     $compile_mode                    = 'off',
-  Array[String]                   $ssl_protocols                   = ['TLSv1', 'TLSv1.1', 'TLSv1.2'],
-  Optional[Array]                 $ssl_cipher_suites               = undef,
-  Boolean                         $enable_profiler                 = false,
-  Pupmod::ProfilingMode           $profiling_mode                  = 'off',
-  Stdlib::AbsolutePath            $profiling_output_file           = "${vardir}/server_jruby_profiling",
-  Array[Simplib::Hostname]        $admin_api_whitelist             = [$facts['fqdn']],
-  String                          $admin_api_mountpoint            = '/puppet-admin-api',
-  Boolean                         $log_to_file                     = false,
-  Boolean                         $syslog                          = simplib::lookup('simp_options::syslog', { 'default_value' => false }),
-  String                          $syslog_facility                 = 'LOCAL6',
-  String                          $syslog_message_format           = '%logger[%thread]: %msg',
-  Pupmod::LogLevel                $log_level                       = 'WARN',
-  Optional[Array[String[1]]]      $autosign_hosts                  = undef,
-  String                          $package_ensure                  = 'latest',
-  Boolean                         $mock                            = false
-) inherits ::pupmod::params {
+  Simplib::IP                                         $bind_address                    = '0.0.0.0',
+  Simplib::IP                                         $ca_bind_address                 = '0.0.0.0',
+  Boolean                                             $auditd                          = simplib::lookup('simp_options::auditd', { 'default_value' => false }),
+  Simplib::Port                                       $ca_port                         = simplib::lookup('simp_options::puppet::ca_port', { 'default_value' => 8141 }),
+  Simplib::NetList                                    $trusted_nets                    = simplib::lookup('simp_options::trusted_nets', { 'default_value' => ['127.0.0.1','::1'] }),
+  String                                              $server_distribution             = simplib::lookup('simp_options::puppet::server_distribution', { 'default_value' => 'PC1' } ),
+  Pupmod::CaTTL                                       $ca_ttl                          = '10y',
+  Boolean                                             $daemonize                       = true,
+  Boolean                                             $enable_ca                       = true,
+  Boolean                                             $enable_master                   = true,
+  Stdlib::AbsolutePath                                $environmentpath                 = $::pupmod::environmentpath,
```

Comment:
What if pupmod class is not included?  I thought we were *not* supposed to set class parameters to values that may not exist?

Reply:
This change should be reverted.

Reply:
Did not see inheritance change.  This is not a problem.